feat: remove 32-bit linux from auto arch, fix auto32 on linux aarch64 (#2458)
pypa/cibuildwheel: docs/options.md

- Windows: `AMD64` `x86` `ARM64`
- Pyodide: `wasm32`
- iOS: `arm64_iphoneos` `arm64_iphonesimulator` `x86_64_iphonesimulator`
- `auto`: The recommended archs for your machine - see the table below.

>     - `auto64`: Just the 64-bit auto arch(s) (include device and simulator for iOS)
>     - `auto32`: Just the 32-bit auto arch

Reviewer: These need their own definition now they're no longer subsets of `auto`.

```suggestion
- `auto64`: The 64-bit arch(s) supported by your machine (includes device and simulator for iOS)
- `auto32`: The 32-bit arch supported by your machine
```
